### PR TITLE
rustdoc: Cleanup `DocFragment`

### PR DESCRIPTION
**Pull request by camelid**
- Remove unused `DocFragment.line` field
- Avoid using `Iterator::count()` where possible

### CODE REVIEW

**Comment by jyn514** [comment]
r=me whether or not there are perf improvements

File: src/librustdoc/html/markdown.rs [hunk 1]
```diff
@@ -656,7 +656,7 @@ impl<'a, I> Footnotes<'a, I> {
     }
 
     fn get_entry(&mut self, key: &str) -> &mut (Vec<Event<'a>>, u16) {
-        let new_id = self.footnotes.keys().count() + 1;
+        let new_id = self.footnotes.len() + 1;
```
Comment (camelid) [inline src/librustdoc/html/markdown.rs:659]: There's no possible behavior change here, right?

Comment (jyn514) [reply src/librustdoc/html/markdown.rs:659]: I can't imagine one, no.

**Comment by camelid** [comment]
Up to -0.36% on Serde instructions and =0.62% on Serde max-rss. Regressions are noise.

cc @rylev (do you want me to still ping you when there are significant spurious results?)

@rustbot label -perf-regression

File: src/librustdoc/passes/unindent_comments.rs [hunk 2]
```diff
@@ -87,7 +87,7 @@ fn unindent_fragments(docs: &mut Vec<DocFragment>) {
     };
 
     for fragment in docs {
-        if fragment.doc.as_str().lines().count() == 0 {
+        if fragment.doc.as_str().lines().next().is_none() {
```
Comment (notriddle) [inline src/librustdoc/passes/unindent_comments.rs:90]: Is there any case where this isn’t equivalent to `fragment.doc.as_str() == “”`?

Comment (camelid) [reply src/librustdoc/passes/unindent_comments.rs:90]: I wasn't sure, but I don't think so. Do you want to ask in `#general` or somewhere else and open a followup PR making that change?

Comment (jyn514) [reply src/librustdoc/passes/unindent_comments.rs:90]: Yeah this should just check if it's empty, I'd rather make the change here.
```suggestion
        if fragment.doc == kw::Empty {
```

**Comment by jyn514** [comment]
@bors r-

https://github.com/rust-lang/rust/pull/91034#discussion_r753306212

**Comment by jyn514** [comment]
Max-rss improvement on this is actually pretty decent, between -.8% and -1.4% :)

**Pull request merged**